client/components/chart/utils.js

```diff
@@ -187,6 +204,74 @@ export const getLine = ( xLineScale, yScale ) =>
 		.x( d => xLineScale( new Date( d.date ) ) )
 		.y( d => yScale( d.value ) );
 
+/**
+ * Describes getXTicks
+ * @param {array} uniqueDates - all the unique dates from the input data for the chart
+ * @param {integer} width - calculated page width
+ * @param {string} layout - standard, comparison or compact chart types
+ * @returns {integer} number of x-axis ticks based on width and chart layout
+ */
+export const getXTicks = ( uniqueDates, width, layout ) => {
+	// caluclate the maximum number of ticks allowed in the x-axis based on the width
+	// and layout of the chart
+	let ticks = 16;
+	if ( width < 783 ) {
+		ticks = 7;
+	} else if ( width >= 783 && width < 1129 ) {
+		ticks = 12;
+	} else if ( width >= 1130 && width < 1365 ) {
+		if ( layout === 'standard' ) {
+			ticks = 16;
+		} else if ( layout === 'comparison' ) {
+			ticks = 12;
+		} else if ( layout === 'compact' ) {
+			ticks = 7;
+		}
+	} else if ( width >= 1365 ) {
+		if ( layout === 'standard' ) {
+			ticks = 31;
+		} else if ( layout === 'comparison' ) {
+			ticks = 16;
+		} else if ( layout === 'compact' ) {
+			ticks = 12;
+		}
+	}
+	if ( uniqueDates.length <= ticks ) {
+		return uniqueDates;
+	}
+	let factors = [];
+	let i = 0;
+	//  first we get all the factors of the length of the uniqieDates array
+	// if the number is a prime number or near prime (with 3 factors) then we
+	// step down by 1 integer and try again
+	while ( factors.length <= 3 ) {
+		factors = getFactors( uniqueDates.length - ( 1 + i ) );
+		i += 1;
+	}
+	let newTicks = [];
+	let factorIndex = 0;
+	// newTicks is the first tick plus the smallest factor (initiallY) etc.
+	// however, if we still end up with too many ticks we look at the next factor
+	// and try again unttil we have fewer ticks than the max
+	while ( newTicks.length > ticks || newTicks.length === 0 ) {
```

Comment:
I didn't do extensive testing, but I think it could be possible to avoid the nested loops, here.
```ES6
	let newTicks = [];
	let factorIndex = 0;
	while ( newTicks.length > ticks || newTicks.length === 0 ) {
		if ( newTicks.length > ticks ) {
			factorIndex += 1;
			newTicks = [];
		}
		for ( let idx = 0; idx < uniqueDates.length; idx = idx + factors[ factorIndex ] ) {
			newTicks.push( uniqueDates[ idx ] );
		}
	}
```
should be equivalent to:
```ES6
	const factor = factors.find( f => uniqueDates.length / f < ticks );
	const newTicks = [];
	for ( let idx = 0; idx < uniqueDates.length; idx = idx + factor ) {
		newTicks.push( uniqueDates[ idx ] );
	}
```